pymc_marketing/mmm/base.py

```diff
@@ -336,19 +342,26 @@ def plot_posterior_predictive(
             )
 
             ax.fill_between(
-                x=self.X[self.date_column],
+                x=posterior_predictive_data.date,
```

Comment:
What is the reason for this change? The date column can have a different name.

Reply:
This enables OOS posterior predictive. `self.X` does not get updated upon `self._data_setter(X_new)`. So if the posterior predictive is for new OOS data, then `self.X[self.date_column]` will still be the in-sample dates.